src/main/java/com/hubspot/jinjava/interpret/Context.java

```diff
@@ -526,4 +551,13 @@ public void addDependencies(SetMultimap<String, String> dependencies) {
   public SetMultimap<String, String> getDependencies() {
     return this.dependencies;
   }
+
+  public boolean isProtectedMode() {
```

Comment:
what does this do?

Reply:
It's (going to be) used to signify that certain tags need to have different functionality. For example when executing a `SetTag`, it can normally just be executed and affect the variable on the context. However, if we are evaluating within an IfTag where the correct branch path is unknown due to a deferred value, then if a value is replaced or updated during this "protected" execution mode, then that has to be handled specially.